common/containers/Tabs/GenerateWallet/components/SaveWallet.jsx

```diff
@@ -85,4 +106,18 @@ export default class SaveWallet extends Component {
       </div>
     );
   }
+
+  getBlob() {
+    return makeBlob("text/json;charset=UTF-8", this.keystore);
+  }
+
+  getFilename() {
+    const ts = new Date();
+    return [
+      "UTC--",
+      ts.toJSON().replace(/:/g, "-"),
+      "--",
+      this.props.wallet.getAddress()
+    ].join("");
+  }
```

Comment:
I feel like these should have stayed inside of some kind of library (Potentially the wallet class) rather than being tacked onto a component.